### PR TITLE
web: make a few variables non-null

### PR DESCRIPTION
**Pull request by yjbanov**
## Description

A minor null safety clean-up in `alarm_clock.dart`, `canvaskit_canvas.dart`, `fonts.dart`, `text_editing.dart`, and `window.dart`.

### CODE REVIEW

File: lib/web_ui/lib/src/engine/window.dart [hunk 1]
```diff
@@ -643,20 +643,20 @@ class EngineWindow extends ui.Window {
       case 'flutter/navigation':
         const MethodCodec codec = JSONMethodCodec();
         final MethodCall decoded = codec.decodeMethodCall(data);
-        final Map<String, dynamic>? message = decoded.arguments;
+        final Map<String, dynamic> message = decoded.arguments ?? const <String, dynamic>{};
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/window.dart:646]: @mdebbar Not sure if I should just do `as Map<String, dynamic>` instead. I imagine we always get a map?

File: lib/web_ui/lib/src/engine/text_editing/text_editing.dart [hunk 2]
```diff
@@ -1450,19 +1450,16 @@ class HybridTextEditing {
...
   /// This field is populated using [useCustomEditableElement]. If `null` the
   /// [_defaultEditingElement] is used instead.
   DefaultTextEditingStrategy? _customEditingElement;
 
-  DefaultTextEditingStrategy? get editingElement {
-    if (_customEditingElement != null) {
-      return _customEditingElement;
-    }
-    return _defaultEditingElement;
+  DefaultTextEditingStrategy get editingElement {
```
Comment (nturgut) [inline lib/web_ui/lib/src/engine/text_editing/text_editing.dart:1461]: Do you think we can use => notation of the getter since it's only one line?

**Review by nturgut — APPROVED** [review]

**Review by cbracken — APPROVED** [review]

File: lib/web_ui/lib/src/engine/window.dart (hunk 1 above)
Comment (mdebbar) [reply lib/web_ui/lib/src/engine/window.dart:646]: Today, yes we always get a map, but I'm not sure if we can safely make assumptions like that with platform messages since there's no type-checking whatsoever.

cc @chunhtai

**Review by mdebbar — APPROVED** [review]

File: lib/web_ui/lib/src/engine/window.dart (hunk 1 above)
Comment (yjbanov) [reply lib/web_ui/lib/src/engine/window.dart:646]: We already cast to `Map?` today. My code isn't changing that part. What I'm wondering is whether we should cast to `Map` instead. In the switch statement, we're null-asserting, so we're not expecting a null value.

Comment (yjbanov) [reply lib/web_ui/lib/src/engine/window.dart:646]: For example, is it possible to get a method call where `decoded.method` is neither `routeUpdated` nor `routeInformationUpdated`, perhaps just to reset `_defaultRouteName`. That would be the only situation I can think of where `message` would be null and we wouldn't crash.

Comment (mdebbar) [reply lib/web_ui/lib/src/engine/window.dart:646]: We could add new methods in the future that don't conform to this contract. But in that case, even `Map?` may fail.

TL;DR: go ahead and cast to `Map`.

Comment (mdebbar) [reply lib/web_ui/lib/src/engine/window.dart:646]: I meant that the message for the new method could be anything. It could be a list, right? Or a number, string, etc.

Comment (chunhtai) [reply lib/web_ui/lib/src/engine/window.dart:646]: I think it is ok to cast it `Map`. We can always change that if we have new contract in the future.

Comment (yjbanov) [reply lib/web_ui/lib/src/engine/window.dart]: Cool. Casting to `Map` then.
